x/ethbridge/types/claim.go

```diff
@@ -3,43 +3,35 @@ package types
 import (
 	"encoding/json"
 	"fmt"
+	oracletypes "github.com/Sifchain/sifnode/x/oracle/types"
 	"strconv"
 
-	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-
-	"github.com/Sifchain/sifnode/x/oracle"
 )
 
-// EthBridgeClaim is a structure that contains all the data for a particular bridge claim
-type EthBridgeClaim struct {
-	EthereumChainID       int             `json:"ethereum_chain_id" yaml:"ethereum_chain_id"`
-	BridgeContractAddress EthereumAddress `json:"bridge_registry_contract_address" yaml:"bridge_registry_contract_address"`
-	Nonce                 int             `json:"nonce" yaml:"nonce"`
-	Symbol                string          `json:"symbol" yaml:"symbol"`
-	TokenContractAddress  EthereumAddress `json:"token_contract_address" yaml:"token_contract_address"`
-	EthereumSender        EthereumAddress `json:"ethereum_sender" yaml:"ethereum_sender"`
-	CosmosReceiver        sdk.AccAddress  `json:"cosmos_receiver" yaml:"cosmos_receiver"`
-	ValidatorAddress      sdk.ValAddress  `json:"validator_address" yaml:"validator_address"`
-	Amount                sdk.Int         `json:"amount" yaml:"amount"`
-	ClaimType             ClaimType       `json:"claim_type" yaml:"claim_type"`
-}
-
 // NewEthBridgeClaim is a constructor function for NewEthBridgeClaim
-func NewEthBridgeClaim(ethereumChainID int, bridgeContract EthereumAddress,
-	nonce int, symbol string, tokenContact EthereumAddress, ethereumSender EthereumAddress,
-	cosmosReceiver sdk.AccAddress, validator sdk.ValAddress, amount sdk.Int, claimType ClaimType,
-) EthBridgeClaim {
-	return EthBridgeClaim{
-		EthereumChainID:       ethereumChainID,
-		BridgeContractAddress: bridgeContract,
+func NewEthBridgeClaim(
+	ethereumChainID int64,
+	bridgeContract EthereumAddress,
+	nonce int64,
+	symbol string,
+	tokenContact EthereumAddress,
+	ethereumSender EthereumAddress,
+	cosmosReceiver sdk.AccAddress,
+	validator sdk.ValAddress,
+	amount sdk.Int,
+	claimType ClaimType,
+) *EthBridgeClaim {
```

Comment:
Why did we make this function return a pointer to a newly allocated EthBridgeClaim instead of just a new object?

We should roll this back to not using pointers.